src/python_tools/robot_broadcast_receiver.py

```diff
@@ -0,0 +1,31 @@
+from software.proto.announcement_pb2 import Announcement
+import socket
+import errno
+from time import time, sleep
+
+receiver = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+receiver.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+receiver.setblocking(False)
+receiver.bind(("", 42020))
+
+announcements = []
+timeout = time() + 4  # 4s
+
+while time() < timeout:
+    try:
+        data = receiver.recv(1024)
+    except socket.error as e:
+        err = e.args[0]
+        if err == errno.EAGAIN or err == errno.EWOULDBLOCK:
+            sleep(0.2)
+            continue
+    else:
+        announcement = Announcement()
+        announcement.ParseFromString(data)
+        if announcement not in announcements:
+            announcements.append(announcement)
```

Comment:
this should be in a function that blocks and returns all the announcement packets

you should call this function wrapped in an `if name == __main__`  block below and print out the collected packets

Reply:
changed